guardian/archivehunter
# Feature/adfs

## What does this change?
Removes Panda in favour of direct oauth2 to our ADFS installation.

This includes:

- removal of associated libraries and refresh components
- complete server-side OAuth exchange, in contrast to pluto.  Due to cookie/localstorage limitations the token is used serverside and authentication is carried out via the Play session token
- adding a new database object to store refresh tokens
- support for extracting user avatar picture from the standard "thumbnailPhoto" and/or "jpegPhoto" claims in the returned JWT
- fixes the refresh-loop problem when the token expires; 

Remove repeated user-profile requests in frontend in favour of a context-based solution that only requests at the mount of the root component

## Migrations required

- Updates to the cloudformation must be deployed or login will fail
- Users are keyed by their email address, so data migration is required to update this.  Migration can be carried out by clicking a button in the About section of the Admin tool.  This only needs to be done once when the deployment has been made and checked.

## How to test
- Log in is immediately different. Click the button and you will get bounced to ADFS which either requests username/password or logs you in immediately depending on whether you already have a valid session in ADFS and/or Kerberos on your workstation.
- Your user avatar is still displayed top-right but should be the picture from your building pass
- All other functionality should be the same

## How can we measure success?
- Unblocks subsequent issues allowing updates to clear Snyk errors
- Eliminate annoying "crash" (actually frontend page-loop) when the access token expires
- 
## Have we considered potential risks?
- Key users will be informed and invited to verify that their logins are working properly.  MM Tech will be available online to answer any queries/problems once rollout occurs


## Images
